File: MPP/MPP Proiect/CSharp/networking/utils/RpcConcurrentServer.cs
using System;
using System.Net.Sockets;
using System.Threading;
using log4net;
using services;

namespace networking.utils{
    public class RpcConcurrentServer : AbsConcurrentServer
    {
        private readonly IServices _service;
        private static readonly ILog logger = LogManager.GetLogger("RpcConcurrentServer");
      
        private ClientRpcReflectionWorker worker;
        public RpcConcurrentServer(string host, int port, IServices service) : base(host, port)
        {
            _service = service; 
            logger.Info("RpcConcurrentServer created");
            Console.WriteLine(@"RpcConcurrentServer created");
        }
        protected override Thread CreateWorker(TcpClient client)
        {
            worker = new ClientRpcReflectionWorker(_service, client);
            return new Thread(new ThreadStart(worker.Run));
        }

    }
}

File: MPP/MPP Proiect/Java/Networking/src/main/java/utils/RpcConcurrentServer.java
package utils;

import rpcprotocol.ClientRpcReflectionWorker;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import service.IService;

import java.net.Socket;


public class RpcConcurrentServer extends AbsConcurrentServer{
    private final IService server;
    private static final Logger logger = LogManager.getLogger();

    public RpcConcurrentServer(int port, IService server){
        super(port);
        this.server = server;
        logger.info("RpcConcurrentServer created");
    }

    @Override
    protected Thread createWorker(Socket client){
        ClientRpcReflectionWorker worker = new ClientRpcReflectionWorker(server, client);
        Thread tw = new Thread(worker);
        return tw;
    }

    @Override
    public void stop(){
        logger.info("Stopping services ...");
    }
}